### ISSUE TITLE
Can't reproduce error rates of VGG-16 using pre-trained model

### ISSUE THREAD

**Issue by malrabeiah** [vlfeat/matconvnet#1147]
Hello everyone,

I downloaded the pre-trained VGG-16 model named "imagenet-matconvnet-vgg-verydeep-16.mat" from: [MatConvNet](uhttp://www.vlfeat.org/matconvnet/pretrained/). Then, using the ILSVRC-2012 dataset, prepared via the cnn_imagenet_setup_data.m script, I tried evaluating the network using the cnn_imagenet_evaluate() script to reproduce the error rates reported in the 2014 paper. and I got some odd results; the top-1 and the top-5 error rates were around 0.998%  and 0.98%, respectively! This is very strange considering that I modified the evaluation script to load the right network and made sure the solver (cnn_train().m) accessed the right validation data--  the modified lines in the script could be found below. It seems like the evaluation script is not properly using either the dataset or the pre-trained model!

Has anyone come across a similar problem? If so, what should I do to reproduce the original results? Do I need to omit some lines in the evaluation script? Or should I do some sort of image pre-processing?

----------------------------------------------
In cnn_imagenet_evaluate(), I modified lines 7-9 as follows:
`opts.dataDir = fullfile(vl_rootnn, 'data', 'imagenet12') ;`
`opts.expDir = fullfile(vl_rootnn, 'data', 'imagenet12-eval-vgg-16') ;`
`opts.modelPath = fullfile(vl_rootnn,'data', 'models', 'imagenet-matconvnet-vgg-verydeep-16.mat') ;`

This, in my case, ensures correct access to the pre-trained model and validation data.

**Comment by malrabeiah** [comment]
I have managed to pinpoint the problem to a bug in the data-preparing script, namely cnn_imagenet_setup_data.m. Line 124:
`ims = dir(fullfile(opts.dataDir, 'images', 'val', '*.JPEG')) ;`
`names = sort({ims.name}) ;`
reads the names of the validation image files into ims. These names are "alphabetically" sorted using "sort", which is the core of the problem; the ground truth labels provided in the "ILSVRC2012_validation_ground_truth.txt" file, found in the "ILSVRC2012_devkit_t12" directory, lists the labels according to a natural order of the image names.

One way to solve the problem is by replacing the MATLAB sort function with a natural-sort function. I used the third-party function provided here:   [natural sort](https://www.mathworks.com/matlabcentral/fileexchange/10959-sort-nat--natural-order-sort)
